### PR TITLE
Feature/jr2 345 create mapper for candidate search request

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

File: src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts [hunk 1]
```diff
@@ -13,33 +13,41 @@ import {
...
         experience: Experience[experience],
-        'residence': residence,
+        residence: mapResidence(searchFilter.residence),
         availability: Availability[availability],
-        'workLoad': workLoad,
+        workLoad: mapWorkLoad(searchFilter.workload),
         workForm: WorkForm[workForm],
         educationLevel: ISCED_1997[educationLevel],
-        graduation: Graduation[graduation],
+        graduation: mapGraduation(graduation),
```
Comment (michael-wirth) [inline src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts]: Why is `Graduation` handled differently than the other enumerations?

Comment (varusakov) [reply src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts]: It's passed differently. It can be passed as string or number

Comment (michael-wirth) [reply src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts]: but why is it passed once as string and once as number? I'd prefer if it we wouldn't mix the values if possilbe

Comment (varusakov) [reply src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts]: I'll try to fix that

Comment (michael-wirth) [reply src/main/webapp/app/candidate-search/state-management/util/search-request-mapper.ts]: Are you still working on this?

**Pull request merged**